ideascube/serveradmin/catalog.py

```diff
@@ -427,26 +440,41 @@ def _progress(*args):
 
     def list_installed(self, ids):
         pkgs = self._get_packages(
-            ids, self._catalog['installed'], fail_if_no_match=False)
+            ids, self._catalog['installed'],
+            fail_if_no_match=False, fail_if_invalid_package=False)
         return sorted(pkgs, key=attrgetter('id'))
 
     def list_available(self, ids):
         pkgs = self._get_packages(
-            ids, self._catalog['available'], fail_if_no_match=False)
+            ids, self._catalog['available'],
+            fail_if_no_match=False, fail_if_invalid_package=False)
         return sorted(pkgs, key=attrgetter('id'))
 
     def list_upgradable(self, ids):
         pkgs = []
 
         for ipkg in self._get_packages(
-                ids, self._catalog['installed'], fail_if_no_match=False):
+                ids, self._catalog['installed'],
+                fail_if_no_match=False, fail_if_invalid_package=False):
             upkg = self._get_package(ipkg.id, self._catalog['available'])
 
             if ipkg != upkg:
                 pkgs.append(upkg)
 
         return sorted(pkgs, key=attrgetter('id'))
 
+    def list_nothandled(self, ids):
```

Comment:
FTR: I have been convinced by the others that even if we had pre-validation of the catalog cache, we'd still need this code to not crash on unhandled package types.

The obvious use case is a server that gets updated, the new code has **dropped** support for an old package type, but the local cache still has packages available with that old type.